depresolve.go

```diff
@@ -214,3 +210,12 @@ func standardRepoHostImportPathToCloneURL(importPath string) (string, error) {
 	}
 	return "https://" + strings.Join(parts[:3], "/") + ".git", nil
 }
+
+func replaceImportPathRepoRoot(importPath, root, cloneURL string) string {
```

Comment:
Add a docstring describing why this is necessary and if it is a bit of a hack, make that explicit.